lang/cmu_is_lex/cmu_is_lex.c

```diff
@@ -0,0 +1,316 @@
+/*  WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN   */
+/*  AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION,          */
+/*  ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF       */
+/*  THIS SOFTWARE.                                                       */
+/*                                                                       */
+/*************************************************************************/
+/*                                                                       */
+/*  is Lexical function                                            */
+/*                                                                       */
+/*************************************************************************/
+#include "flite.h"
+#include "cst_val.h"
+#include "cst_voice.h"
+#include "cst_lexicon.h"
+#include "cst_ffeatures.h"
+#include "cmu_is_lex.h"
+
+extern const int cmu_is_lex_entry[];
+extern const unsigned char cmu_is_lex_data[];
+extern const int cmu_is_lex_num_entries;
+extern const int cmu_is_lex_num_bytes;
+extern const char * const cmu_is_lex_phone_table[71];
+extern const char * const cmu_is_lex_phones_huff_table[];
+extern const char * const cmu_is_lex_entries_huff_table[];
+
+static int cmu_is_is_vowel(const char *p);
+static int cmu_is_is_silence(const char *p);
+static int cmu_is_has_vowel_in_list(const cst_val *v);
+static int cmu_is_has_vowel_in_syl(const cst_item *i);
+static int cmu_is_sonority(const char *p);
+
+// cst_val *cmu_is_lex_lts_function(const struct lexicon_struct *l, 
+//                                    const char *word, const char *pos)
+// {
+//     return NULL;
+// }
+
+static int cmu_is_is_silence(const char *p)
+{
+    if (cst_streq(p,"pau"))
+	return TRUE;
+    else
+	return FALSE;
+}
+
+static int cmu_is_has_vowel_in_list(const cst_val *v)
+{
+    const cst_val *t;
+
+    for (t=v; t; t=val_cdr(t))
+	if (cmu_is_is_vowel(val_string(val_car(t))))
+	    return TRUE;
+    return FALSE;
+}
+
+static int cmu_is_has_vowel_in_syl(const cst_item *i)
+{
+    const cst_item *n;
+
+    for (n=i; n; n=item_prev(n))
+	if (cmu_is_is_vowel(item_feat_string(n,"name")))
+	    return TRUE;
+    return FALSE;
+}
+
+static int cmu_is_is_vowel(const char *p)
+{
+    /* this happens to work for icelandic SAMPA phoneset */
+    if (strchr("aeiouOEYI9",p[0]) == NULL)
+	return FALSE;
+    else
+	return TRUE;
+}
+
+static int cmu_is_sonority(const char *p)
+{
+    /* A bunch of hacks for US English phoneset */
+    if (cmu_is_is_vowel(p) || (cmu_is_is_silence(p)))
+	return 5;
+    else if ((sizeof(p) < 2 || strchr("0",p[2]) == NULL) && strchr("wylr",p[0]) != NULL)
+	return 4;  /* glides/liquids */
+    else if (strchr("nmNJ",p[0]) != NULL)
+	return 3;  /* nasals */
+    else if (strchr("bdgjvzGD",p[0]) != NULL)
+	return 2;  /* voiced obstruents */
+    else
+	return 1;
+}
+
+int cmu_is_syl_boundary(const cst_item *i,const cst_val *rest)
+{
+    // TODO adapt to is
```

Comment:
Are the TODO's in this file real TODO's for our use case we feed only phonemes to this voice ? We will probably never feed raw text to the Flite voice anyway. Can we make sure, that these functions never change the phoneme text or simply make noops or even better: never call these ?

Reply:
If the flite code is used in the same way as when providing the binary with a -p flag then these functions are actually never called.

This is what I was referring to in our meeting today. I can replace the TODO with a UNIMPLEMENTED or something like that if you prefer.

Reply:
Yes unimplemented sounds better. But are the dummy implementations really no-ops or are the stubs just assuming that the implementer does fill in the missing pieces, otherwise it crashes ?

Reply:
Right now, this method causes a segfault when it's called. But we never want it called anyway while it hasn't been implemented

Reply:
Ok, I will probably tidy up the files anyway, when I am implementing the library.